### PR TITLE
feat: add skill for flask quickstart

### PR DESCRIPTION
**Pull request by kishore7snehil**
### Description

Adding basic skill for securing Flask using `auth0-server-python`

### CODE REVIEW

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+---
+
+## Prerequisites
+
+- Flask application
+- Auth0 Regular Web Application configured (not an API — must be an Application)
+- If you don't have Auth0 set up yet, use the `auth0-quickstart` skill first
+
+## When NOT to Use
```
Comment (arpit-jn) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md:21]: The "When NOT to Use" section only mentions API use cases but doesn't provide a routing table to other skills. Every other shipped skill has a structured table with clear routing (see auth0-express, express-oauth2-jwt-bearer, auth0-api-python). 

This skill should route users to the correct alternative for common mismatches.

## Current:

## When NOT to Use
`- **APIs / microservices** — `auth0-server-python` is a webapp SDK...

## Suggestion:

## When NOT to Use
| Use Case | Recommended Skill |
|----------|------------------|
| Python API with JWT Bearer validation (Flask/FastAPI) | `auth0-api-python` |
| FastAPI web app with login/logout UI | `auth0-fastapi` |
| Django web app with login/logout UI | See [Django quickstart](https://auth0.com/docs/quickstart/webapp/django) |
| Express.js web app with login/logout UI | `auth0-express` |
| React / Angular / Vue SPA calling an API | `auth0-react`, `auth0-angular`, `auth0-vue` |
| Node.js API with JWT Bearer validation | `express-oauth2-jwt-bearer` |

This could be helpful for Flask because users commonly confuse Flask web apps (session-based, this skill) with Flask API endpoints (JWT Bearer, auth0-api-python).

**Comment by arpit-jn** [comment]
All recently shipped skills include evaluation infrastructure (`tests/graders.json`, `tests/evals.json`, `tests/prompt.md`, `tests/run-evals.mjs`, `tests/benchmark-config.json`). This skill has no evals, making it difficult to measure skill attribution value or detect regressions.

**Recommendation:** Add a `tests/` directory with at minimum graders.json, evals.json, prompt.md, and benchmark-config.json. The `run-evals.mjs` and `graders.ts` can be copied from another skill and adapted.

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+`start_interactive_login()` returns a URL string pointing to Auth0's Universal Login page. You must wrap it in `redirect()`. Authorization params (scope, redirect_uri) are already configured on the `ServerClient`.
+
+### 7. Add Callback Route
+
+```python
+@app.route("/callback")
+async def callback():
+    try:
+        await auth0.complete_interactive_login(str(request.url))
+        return redirect("/")
```
Comment (arpit-jn) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md:199]: This redirects to / after successful callback:

`return redirect("/")
`
But no step ever defines `@app.route("/")`. A developer following the quickstart exactly will get a 404 after logging in.

**Potential Fix:** Add a step (or append to step 5) with a basic home route:

```
@app.route("/")
async def home():
    user = await auth0.get_user()
    if user:
        return f"Hello, {user['name']}! <a href='/profile'>Profile</a> | <a href='/logout'>Logout</a>"
    return "Welcome! <a href='/login'>Login</a>"
```

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+### 8. Add Profile Route (Protected)
+
+```python
+@app.route("/profile")
+async def profile():
+    user = await auth0.get_user()
+    if user is None:
+        return redirect("/login")
+    return render_template("profile.html", user=user)
```
Comment (arpit-jn) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: Step 8 calls `render_template("profile.html", user=user)` but the quickstart never provides the template. A developer will hit a TemplateNotFoundError at runtime. The user dict keys (sub, name, email, picture, email_verified) are documented in `api.md:142-147` but not referenced from SKILL.md.

**Potential Fix:** Either add a minimal template inline in step 8, or change step 8 to return JSON/plain text like the home route suggestion above, keeping the quickstart simpler.

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+
+**Cookie size note:** Stateless sessions store all data in a cookie (~4KB limit). For most apps this is sufficient. If you store large amounts of session data or hit cookie size limits, switch to [stateful setup](#stateful-setup-with-redis).
+
+### 5. Configure Flask App
+
+In `app.py`, set up Flask with the secret key and session configuration:
+
+```python
+import os
+from flask import Flask, redirect, render_template, request, url_for
```
Comment (arpit-jn) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: This imports `url_for` but no subsequent step uses it. A developer (or linter) would flag this as an unused import.

**Fix:** Remove url_for from the import line in step 5.

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    url = await auth0.logout()
+    return redirect(url)
+```
+
+`logout()` returns the Auth0 logout URL. Redirect the user to it.
+
+### 10. Test the App
+
+```bash
+flask run
```
Comment (arpit-jn) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md:238]: This says `flask run` but never mentions that the file must be named app.py (which it is, so it works by convention) or setting `FLASK_APP`. 
This is minor since app.py is the Flask default, but worth noting for clarity.

**Review by arpit-jn — CHANGES_REQUESTED** [review]

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 5 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md:238]: Step 5 explicitly names the file `app.py` which is Flask's default discovery convention , no `FLASK_APP` needed. Adding a note would over-explain standard Flask behavior.

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 4 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: Addressed in [1ba144d](https://github.com/auth0/agent-skills/pull/48/changes/1ba144d988b10bcdf655fd68a24d975968d52b0f)

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 3 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: Addressed in [1ba144d](https://github.com/auth0/agent-skills/pull/48/changes/1ba144d988b10bcdf655fd68a24d975968d52b0f)

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 2 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md:199]: Addressed in [1ba144d](https://github.com/auth0/agent-skills/pull/48/changes/1ba144d988b10bcdf655fd68a24d975968d52b0f)

**Comment by kishore7snehil** [comment]
> All recently shipped skills include evaluation infrastructure (tests/graders.json, tests/evals.json, tests/prompt.md, tests/run-evals.mjs, tests/benchmark-config.json). 

Addressed in [18a4b1e](https://github.com/auth0/agent-skills/pull/48/commits/18a4b1ef94593a4b29c2e3609842b143eaec9ca0)

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+
+```python
+import os
+from flask import session as flask_session
+from auth0_server_python.auth_server.server_client import ServerClient
+from auth0_server_python.auth_types import StateData, TransactionData
+from auth0_server_python.store import StateStore, TransactionStore
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (nandan-bhat) [inline plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: By default `load_dotenv()` loads `.env` files. Should we add a comment suggesting to adjust the path if the customer has set `.env.local` ?

File: plugins/auth0-sdks/skills/auth0-flask/references/integration.md [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+@app.route("/api-call")
+async def api_call():
+    user = await auth0.get_user()
+    if user is None:
+        return redirect("/login")
+
+    access_token = await auth0.get_access_token()
+
+    if access_token is None:
+        return "No access token available", 401
```
Comment (nandan-bhat) [inline plugins/auth0-sdks/skills/auth0-flask/references/integration.md]: `auth0-server-python` would return `AccessTokenError ` if AT is missing.

- https://github.com/auth0/auth0-server-python/blob/main/src/auth0_server_python/auth_server/server_client.py#L958

This block will never be hit.

File: plugins/auth0-sdks/skills/auth0-flask/references/integration.md [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+
+### Callback Error Handling
+
+```python
+@app.route("/callback")
+async def callback():
+    try:
+        await auth0.complete_interactive_login(str(request.url))
+        return redirect("/")
+    except ValueError as e:
```
Comment (nandan-bhat) [inline plugins/auth0-sdks/skills/auth0-flask/references/integration.md]: If there's state mismatch, the SDK `auth0-server-python` would return `MissingRequiredArgumentError` or `MissingTransactionError`.

- https://github.com/auth0/auth0-server-python/blob/main/src/auth0_server_python/auth_server/server_client.py#L565
- https://github.com/auth0/auth0-server-python/blob/main/src/auth0_server_python/auth_server/server_client.py#L2023

File: plugins/auth0-sdks/skills/auth0-flask/references/api.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+
+### get_access_token()
+
+Retrieves the access token for calling external APIs. Handles token refresh automatically if a refresh token is available and the access token is expired.
+
+```python
+access_token = await auth0.get_access_token()
+```
+
+**Returns:** Access token string or `None`.
```
Comment (nandan-bhat) [inline plugins/auth0-sdks/skills/auth0-flask/references/api.md]: Can we confirm this ?

SDK seems to return `AccessTokenError`

- https://github.com/auth0/auth0-server-python/blob/main/src/auth0_server_python/auth_server/server_client.py#L958

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 6 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: `load_dotenv()` loading `.env `is standard Python convention. Adding `.env.local `guidance would be over-explaination.

File: plugins/auth0-sdks/skills/auth0-flask/references/integration.md (hunk 8 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/references/integration.md]: The except `Exception as e `catch-all already handles all SDK errors regardless of class. This is a quickstart skill ,  we intentionally use generic error handling rather than importing SDK-specific exception types.

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 6 above)
Comment (nandan-bhat) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: Why ?
You are already mentioning about it in multiple places.
https://github.com/auth0/agent-skills/blob/eb144841041228bd73451c576b743d93dab78c67/plugins/auth0-sdks/skills/auth0-flask/references/setup.md?plain=1#L26

Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: The difference here is context -> SKILL.md is the quickstart walkthrough where we show load_dotenv() as the standard pattern. The setup.md script handles file detection because it writes credentials and needs to know where. But the SKILL.md code example is showing how to read them and load_dotenv() loading .env is the default path that works for the manual quickstart. If the user has .env.local, they'd pass it explicitly: load_dotenv(".env.local")

File: plugins/auth0-sdks/skills/auth0-flask/references/api.md (hunk 9 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/references/api.md]: Addressed in [71d68a7](https://github.com/auth0/agent-skills/pull/48/changes/71d68a7f33132640fdb420371b996dd4b5087528)

File: plugins/auth0-sdks/skills/auth0-flask/references/integration.md (hunk 7 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/references/integration.md]: The idea here was to have a generic error wrapping but you are right about the code block never getting hit.


Addressed in [71d68a7](https://github.com/auth0/agent-skills/pull/48/changes/71d68a7f33132640fdb420371b996dd4b5087528)

**Review by arpit-jn — DISMISSED** [review]

File: plugins/auth0-sdks/skills/auth0-flask/SKILL.md (hunk 6 above)
Comment (kishore7snehil) [reply plugins/auth0-sdks/skills/auth0-flask/SKILL.md]: Addressed in [35f1fa0](https://github.com/auth0/agent-skills/pull/48/commits/35f1fa0773b17e211e4168dcb4243f16234f108f)

**Review by nandan-bhat — DISMISSED** [review]

**Review by nandan-bhat — APPROVED** [review]

**Review by arpit-jn — APPROVED** [review]
